### PR TITLE
Additional checks for Pulsar Function jar submission

### PR DESCRIPTION
**Pull request by lucperkins**
This PR adds more fine-tuned error reporting for submitting Pulsar Function jars. There are now specific errors provided when:

* The specified path doesn't exist at all
* The specified path is a directory
* The file isn't a `*.jar` file

At the moment, the only feedback that's available is that the jar doesn't contain the specified class, which is misleading in many cases.

### CODE REVIEW

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdFunctions.java [hunk 1]
```diff
@@ -249,15 +251,31 @@ void processArguments() throws Exception {
         }
 
         private void doJavaSubmitChecks(FunctionConfig.Builder functionConfigBuilder) {
-            File file = new File(jarFile);
+            if (Files.isDirectory(Paths.get(jarFilePath))) {
+                throw new IllegalArgumentException(String.format(
+                        "The path %s is a directory, not a jar file", jarFilePath));
+            }
+
+            if (!jarFilePath.endsWith(".jar")) {
```
Comment (sijie) [inline pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdFunctions.java]: not sure we need this?

**Comment by srkukarni** [comment]
retest this please

**Comment by lucperkins** [comment]
@jerrypeng @sijie I agree that that's not a necessary check, so I've updated it with a more robust check

**Comment by merlimat** [comment]
@lucperkins There are failures in the tests: https://builds.apache.org/job/pulsar_precommit_java8/71/

**Comment by lucperkins** [comment]
retest this please

**Comment by lucperkins** [comment]
I don't have the bandwidth to figure out why the tests are failing at the moment, so I'm going to close this for now and re-open later (in the interest of clearing out noise in the backlog).

**Pull request closed without merging**